testing/scripts/playwright_test.py

```
@@ -10532,6 +10532,207 @@ async def test_v371_footer_focus_ring(page: Page) -> None:
+async def test_v380_ipv4_tool_groups(page: Page) -> None:
+    """v3.8.0 #441 — IPv4 tools organised into 3 labelled groups: Transform / Visualize / Lookups."""
+    section("v3.8.0 #441 — IPv4 tool IA grouping")
+    await navigate(page, APP_URL)
+
+    labels = await page.eval_on_selector_all(
+        "#panel-ipv4 .tool-toolbar-group-label",
+        "els => els.map(el => el.textContent.trim())",
+    )
+    assert_eq("3 group labels present", len(labels), 3)
+    assert_eq("group labels match spec",
+              [l.lower() for l in labels],
```

Ruff (0.15.13)

[error] Ambiguous variable name: `l`

(E741)